Sources/FoundationXML/XMLParser.swift

```diff
@@ -575,9 +588,33 @@ open class XMLParser : NSObject {
 
         return result
     }
-    
+#endif
+
+    internal func parse(from data: Data) -> Bool {
+        var data = data
+        var result = true
+        let buffer = malloc(_chunkSize)!.bindMemory(to: UInt8.self, capacity: _chunkSize)
+        defer { free(buffer) }
+        var range = NSRange(location: 0, length: min(_chunkSize, data.count))
+        while result {
+            let chunk = data.withUnsafeMutableBytes { (rawBuffer: UnsafeMutableRawBufferPointer) -> Data in
+                let ptr = rawBuffer.baseAddress!.advanced(by: range.location)
+                return Data(bytesNoCopy: ptr, count: range.length, deallocator: .none)
```

Comment:
This escapes a scoped inner-pointer. Since Data is self-slicing, can't you just use a regular slice?